sjsonnet/src/sjsonnet/ValScope.scala

```diff
@@ -18,23 +18,11 @@ final class ValScope private (val bindings: Array[Lazy]) extends AnyVal {
 
   def length: Int = bindings.length
 
-  def extend(
-      newBindingsF: Array[(Val.Obj, Val.Obj) => Lazy] = null,
-      newSelf: Val.Obj,
-      newSuper: Val.Obj): ValScope = {
-    val by = if (newBindingsF == null) 2 else newBindingsF.length + 2
-    val b = Arrays.copyOf(bindings, bindings.length + by)
+  def extend(newBindings: Array[Lazy], newSelf: Val.Obj, newSuper: Val.Obj): ValScope = {
+    val b = Arrays.copyOf(bindings, bindings.length + newBindings.length + 2)
```

Comment:
bindings.length and newbindings.length can be extracted to a local variable?

Reply:
I would prefer to leave this as-is for now, given that the other methods (including `extendSimple`) have similar repetition of `.length`. My understanding is that JIT should already be optimizing this nicely.